### ISSUE TITLE
[css-shapes-2] In `shape()` It should be possible to specify control points relative to their point on the curve

### ISSUE THREAD

**Issue by smfr** [w3c/csswg-drafts#10667]
In CSS Basic Shapes 2, the `[shape()](https://drafts.csswg.org/css-shapes-2/#shape-function)` function allows you specify points with `to` or `by` which indicates whether the point, _and its control points_, are specified in "absolute" terms relative to the bounding box, or in "relative" terms from the current path point.

However, in many cases I think authors will want to specify control points relative to the path point that they are attached to. I should be able to say something like `curve to 120px 140px with -12px -10px / 15px 20px` to say that the first control point is at `-12px -10px` relative to the start of the segment, and the second one is at `15px 20px` relative to the end point.

We'd need some reasonable behavior for quadratic segments, and smooth segments.

**Comment by smfr** [comment]
Another enhancement is to allow authors to specify control point positions as percentages of their path segment, which means we have two ways to resolve percentages on control points: relative to the reference box, and relative to the "segment box" (a box enclosing the two end points of the segment).

![control point options](https://github.com/user-attachments/assets/d1cb423e-9571-4eb5-86d2-10c12ccf678d)

**Comment by smfr** [comment]
If we allow segment-relative percentages, authors will need to be able to mix and match for the two axes on a control point (say, using segment-% for the horizontal component, and box-% for the vertical), since a segment's start and end point can be at the same X or Y position.

**Comment by smfr** [comment]
@noamr

**Comment by noamr** [comment]
For the latter point, perhaps it can be a type of unit? `20segh`

**Comment by noamr** [comment]
Perhaps:
 `curve to 120px 140px with -12px -10px from start / 15px 20px from end`
 
 This can also allow mixing relative/absolute:
`curve by 200px 200px with -20px 20px from top left`
`smooth to bottom right with 30px 30px from segment-start`

**Comment by tabatkins** [comment]
> or the latter point, perhaps it can be a type of unit? 20segh

Yeah, if we do this it needs to be a new unit. It's not a referencing mode at all; it can be meaningful to use in either absolute or relative positioning, and mixed with normal %s.

**Comment by smfr** [comment]
I like `segh`, `segv`, where 100segh is 100% of the horizontal distance between the segment start and end points.

**Comment by smfr** [comment]
I think we could also simplify things by collapsing the different segment types by using `auto`. For example, `hline` could 
be:

```
line to auto, 30px
```

 and `vline` could be:

```
line to 50px, auto
```

We could also collapse `smooth` into `curve`:

The smooth cubic curve `smooth to 120px 140px with 150px 200px` could be:
```
curve to 120px 140px with auto / 150px 200px
```

and the smooth quadratic curve `smooth to 120px 140px` could be:
```
curve to 120px 140px with auto
```

**Comment by noamr** [comment]
> I think we could also simplify things by collapsing the different segment types by using `auto`. For example, `hline` could be:
> 
> ```
> line to auto, 30px
> ```
> 
> and `vline` could be:
> 
> ```
> line to 50px, auto
> `


It was like this in the original `shape()` proposal, I think `vline` and `hline` are more readable.

**Comment by noamr** [comment]
Removed the "Needs edit", but leaving this open for the `segw` `segh` proposal.
